dev-client/src/hooks/index.tsx

```diff
@@ -0,0 +1,22 @@
+import {useEffect, useState} from 'react';
+import {Keyboard} from 'react-native';
+
+export const useKeyboardOpen = () => {
+  const [keyboardOpen, setKeyboardOpen] = useState<boolean>(
+    Keyboard.isVisible(),
+  );
+  useEffect(() => {
+    const openListener = Keyboard.addListener('keyboardDidShow', () =>
+      setKeyboardOpen(true),
+    );
+    const closeListener = Keyboard.addListener('keyboardDidHide', () =>
+      setKeyboardOpen(false),
+    );
+    return () => {
+      if (openListener) openListener.remove();
+      if (closeListener) closeListener.remove();
+    };
+  });
```

Comment:
nit: i think without the empty dependency list we will be adding and removing the listeners on each render
```suggestion
  }, []);
```